indico/modules/events/timetable/views/weeks.py

```diff
@@ -0,0 +1,91 @@
+from collections import defaultdict, OrderedDict
+from datetime import timedelta
+from itertools import takewhile
+
+from flask import render_template, request
+from pytz import timezone
+
+from indico.modules.events.timetable.models.entries import TimetableEntryType
+from indico.util.date_time import iterdays
+
+
+def _flatten_timetable(entries):
+    for entry in entries:
+        if entry.type == TimetableEntryType.SESSION_BLOCK:
+            for sub_entry in entry.children:
+                yield sub_entry
+        else:
+            yield entry
+
+
+def _localized_time(dt, tz):
+    return dt.astimezone(tz).time()
+
+
+def inject_week_timetable(event, days, tz_name):
+    first_week_day = request.args.get('first_week_day', 'monday')  # monday/sunday/event
+    sunday_first = (first_week_day == 'sunday')
+    show_end_times = request.args.get('showEndTimes') == '1'
+
+    tz = timezone(tz_name)
+
+    day_dict = defaultdict(list, [(d, i) for d, i in days])
```

Comment:
Yeah, that used to be a list, then a generator... long story. Thanks for spotting.